packages/devextreme-react/src/core/__tests__/props-updating.test.tsx

```diff
@@ -717,6 +717,46 @@ describe('cfg-component option control', () => {
     expect(Widget.option.mock.calls[1]).toEqual(['nestedOption.b', 'const']);
   });
 
+  it('invokes option change guard handlers', () => {
+    const TestContainer = ({ value }: { value: number }) => {
+      return (
+        <ControlledComponent>
+          <NestedComponent a={value} />
+        </ControlledComponent>
+      );
+    };
+
+    const { rerender } = render(<TestContainer value={123} />);
+
+    fireOptionChange('nestedOption.a', 234);
+
+    rerender(<TestContainer value={123} />);
+    jest.runAllTimers();
+
+    expect(Widget.option).toHaveBeenCalledWith('nestedOption.a', 123);
+  });
+
+  it('invokes option change guard handlers in strict mode', () => {
+    const TestContainer = ({ value }: { value: number }) => {
+      return (
+        <React.StrictMode>
```

Comment:
I guess we can leave `<React.StrictMode>`, just please note a different way to activate the Strict Mode in tests:

https://github.com/DevExpress/DevExtreme/blob/24_1/packages/devextreme-react/src/core/__tests__/component.test.tsx#L60

Although it's maybe a little bit more verbose because it requires the option to be reset to `false` on `afterEach`.

And this new syntax wasn't available before `24_1` because I just recently updated the `@testing-library/react` dependency in `24_1`

Reply:
This is noted.